src/stage-options.ts

```diff
@@ -18,7 +18,7 @@ export interface AddGitHubStageOptions extends AddStageOpts {
    *
    * @default - no GitHub environment
    */
-  readonly gitHubEnvironment?: string;
+  readonly gitHubEnvironment?: string | GithubEnvironment;
```

Comment:
I don't think we are allowed to use union types here. I wonder how this got past jsii, and I suspect it probably didn't. At any rate, I think this is one of those cases where we thank our lucky stars we're an "experimental" module still, and break the API. What we want is something like this:

```ts
export interface GitHubEnvironment: {
  readonly name: string;
  readonly url?: string;
}
```

And then users use it like:

```ts
githubEnvironment: { name: 'blah' };
```

or 

```ts
gitHubEnvironment: { name: 'blah', url: 'blah.com' };
```

Then, just for completeness sake, we probably want to have a quick rendering step to make sure that `GitHubEnvironments` with only a name are formatted the way GitHub wants:

```ts
// here we can have a union type. Only in public APIs we cannot, due to jsii limitations.
renderGitHubEnvironment(env: GitHubEnvironment): string | GitHubEnvironment {
  if (env.url === undefined) {
    return: env.name;
  }
  return env;
}

Reply:
Awesome, I always forget about the JSII limits on typescript.